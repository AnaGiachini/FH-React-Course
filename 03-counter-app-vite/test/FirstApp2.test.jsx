import React from 'react';
import { FirstApp } from '../src/FirstApp.jsx';
import { render, screen } from '@testing-library/react';



  describe('Pruebas en FirstApp2', () => {

    const title = 'Ana Giachini';
    const subtitle = 'No hay sub';
  
    test('debe hacer match con el snapshot', () => {
  
      const { container } = render(<FirstApp title={ title } />);
      expect(container).toMatchSnapshot();
  
    });
  
    test('debe mostrar el titulo', () => {
  
      render(<FirstApp title={ title } />);
      expect( screen.getByText( title ) ).toBeTruthy();
  
    });
  
    test('debe mostrar el titulo en h1', () => {
      render(<FirstApp title={ title } />);
      expect( screen.getByRole('heading', { level: 1}).innerHTML ).toContain( title );
    });

    test('debe mostrar el subtitulo', () => {
      render(<FirstApp subtitle={ subtitle } />);
      expect( screen.getAllByText( subtitle ).length ).toBe(1);
    });

  });
