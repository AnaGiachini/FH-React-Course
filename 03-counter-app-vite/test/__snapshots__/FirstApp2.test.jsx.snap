// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp2 debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Ana Giachini
  </h1>
  <p>
    No hay sub
  </p>
  <p />
</div>
`;
